cardano-node/test/Test/Common/Network.hs

```diff
@@ -0,0 +1,49 @@
+module Test.Common.Network
+  ( isPortOpen
+  , listenOn
+  ) where
+
+import           Control.Monad
+import           Control.Monad.Fail
+import           Control.Monad.IO.Class (MonadIO, liftIO)
+import           Data.Bool
+import           Data.Either
+import           Data.Eq
+import           Data.Function
+import           Data.Functor
+import           Data.Int
+import           Data.Maybe
+import           Foreign.C.Error (Errno (..), eCONNREFUSED)
+import           Network.Socket (Family (AF_INET), SockAddr (..), Socket, SocketType (Stream))
+import           System.IO (IO)
+import           Text.Show (show)
+
+import qualified GHC.IO.Exception as IO
+import qualified Network.Socket as IO
+import qualified UnliftIO.Exception as IO
+
+-- | Check if a TCP port is open
+isPortOpen :: Int -> IO Bool
+isPortOpen port = do
+  socketAddressInfo:_ <- IO.getAddrInfo Nothing (Just "127.0.0.1") (Just (show port))
+  canConnect (IO.addrAddress socketAddressInfo)
+
+-- | Check if it is possible to connect to a socket address
+canConnect :: SockAddr -> IO Bool
+canConnect sockAddr = IO.bracket (IO.socket AF_INET Stream 6) IO.close' $ \sock -> do
+  result <- IO.try $ IO.connect sock sockAddr
+  case result of
+    Right () -> return True
+    Left e -> if (Errno <$> IO.ioe_errno e) == Just eCONNREFUSED
+      then return False
+      else IO.throwIO e
```

Comment:
If you want the tests to also work on windows, then be aware that the network exceptions are different.
We have similar code in cardano-wallet:
https://github.com/input-output-hk/cardano-wallet/blob/5c1240cc668993fcb7135f244f2b1cc9c96a3238/lib/core/src/Cardano/Wallet/Network/Ports.hs#L104

Reply:
Your suggestion worked a charm.  Thanks!